LIFX: support for multizone

amelchio opened the pull request:
## Description:

Support multizone lights like the LIFX Z strip. This is available through `lifx_set_state` and allows setting each zone to a different color/brightness. Unfortunately, I do not see a way to make this work with scenes.

This implementation also fixes an issue where adjusting the brightness in the UI would make all zones the same color.

The first commits rearrange things, making it quite simple to add the multizone support in the final commit.

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** home-assistant/home-assistant.github.io#2804

## Example entry for `configuration.yaml` (if applicable):
```yaml
script:
  red_white_blue:
    sequence:
      - service: light.lifx_set_state
        data:
          entity_id: light.z
          color_name: 'red'
          brightness_pct: 80
          transition: 10
          zones: [0,3,6]
      - service: light.lifx_set_state
        data:
          entity_id: light.z
          kelvin: 5600
          brightness_pct: 60
          transition: 10
          zones: [1,4,7]
      - service: light.lifx_set_state
        data:
          entity_id: light.z
          color_name: 'blue'
          brightness_pct: 80
          transition: 10
          zones: [2,5]
```

## Checklist:

If user exposed functionality or configuration variables are added/changed:
  - [X] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [X] Local tests with `tox` run successfully.
  - [X] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [X] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [X] New dependencies have been added to `requirements_all.txt` by running`script/gen_requirements_all.py`.
  - [ ] <s>New files were added to `.coveragerc`.</s>

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

pvizeli commented:
I know that is not relevant to this PR but you had done a good job on refactory of lifx code 👍

amelchio committed:
Use aiolifx features_map for deciding bulb features

Also move the feature detection out of Light so it is available even
during the initial detection.

amelchio committed:
Simplify AwaitAioLIFX

This has become possible with recent aiolifx that calls the callback even
when a message is lost.

Now the wrapper can be used also before a Light is added though the register
callback then has to become a coroutine.

amelchio committed:
Add support for multizone

This lets lifx_set_state work on individual zones.

Also update to aiolifx_effects 0.1.1 that restores the state for individual
zones.

amelchio committed:
Refactor send_color

amelchio committed:
Make aiolifx modules easily available

amelchio committed:
Move each LIFX light type to a separate class

amelchio commented:
Rebased on current `dev`
